Autopilot/Inc/IMU.hpp

```diff
@@ -37,33 +43,45 @@ class IMU{
         virtual void GetResult(IMUData_t &Data) = 0; //
 };
 
-/***********************************************************************************************************************
- * Derived classes
- **********************************************************************************************************************/
-
 //To be replaced with implementation of actual sensor
 class ICM20602: public IMU{
     public:
-        /**
-         * Initializes IMU
-         * */
-        void Init(){}; 
+        ICM20602(const ICM20602*) = delete;
+        static ICM20602* GetInstance();
```

Comment:
THere's a much more c++y way of doing this it turns out. if we create a static object and return a reference to it,then we need not think about pointers. Don't do this now though. I'd like to change this eventually, but we'll change it for all the sensors one shot before their driver implementations are merged.